### PR TITLE
Replace identification of processes with a random nonce instead of using pid and ppid

### LINKED ISSUE #24: Timeout error appears without even 10 seconds passing
I'm at commit: `78abe792f681c2ffd0b3cb813002feadba300a36`
Renpy is latest stable (`Ren'Py 8.2.3.24061702`)

I'm in linux.

When I activate the warp functionality (<kbd>Alt</kbd>+<kbd>Shift</kbd>+<kbd>E</kbd>) on a line of code, it opens Ren'py and successfully connects to the server running in vscode.

However, after connecting and while still connected, the:

> Timed out trying to connect to Ren'Py window. Is the socket client running?

Appears.

The logs from the `autorun.py` script:
```
renpy warp script started
connected to renpy warp socket server
```

Any idea what is really happening here? How can it connect successfully but cause the timeout?

### PR DESCRIPTION
**Pull request by brunoais**
Replace identification of processes with a random nonce instead of using pid and ppid

fixes #24

### CODE REVIEW

File: src/lib/socket.ts [hunk 1]
```diff
@@ -55,21 +55,25 @@ export async function start_websocket_server({
...
-			const rpp = await pm.find_tracked(pid)
+			logger.trace(`renpy socket server ${port} received a connection request with nonce ${req.headers['nonce']}`)
+			const nonce = Number(req.headers['nonce'])
 
-			if (!rpp) throw new Error('no process found for pid ' + pid)
+			const rpp = await pm.find_tracked(nonce)
+
+			if (!rpp){
+				logger.warn(`Rejecting connection to socket because ${nonce} is not registered`);
+				return;
```
Comment (furudean) [inline src/lib/socket.ts:70]:
```suggestion
				return
```
We don't use semicolons in this code base

File: src/lib/process.ts [hunk 2]
```diff
@@ -234,27 +234,17 @@ export class ProcessManager {
 		})
 	}
 
-	get(pid: number): RenpyProcess | undefined {
-		return this.processes.get(pid)
+	get(runningNonce: number): RenpyProcess | undefined {
+		return this.processes.get(runningNonce)
 	}
```
Comment (furudean) [inline src/lib/process.ts:239]: ditto

File: src/lib/process.ts [hunk 3]
```diff
@@ -208,15 +208,15 @@ export class ProcessManager {
...
 
 	/** @param {RenpyProcess} process */
-	async add(process: RenpyProcess) {
+	async add(runningNonce:number, process: RenpyProcess) {
```
Comment (furudean) [inline src/lib/process.ts:211]: ditto

**Review by furudean — CHANGES_REQUESTED** [review]
looks good, just a few small things

**Comment by furudean** [comment]
Please add a line to the CHANGELOG as well please 🙂 And do credit yourself!

File: src/lib/launch.ts [hunk 4]
```diff
@@ -168,10 +168,13 @@ export async function launch_renpy({
 
 			if (strategy === 'Replace Window') pm.kill_all()
 
+			const runningNonce = Math.trunc(Math.random() * 2047483648);
```
[the remark this answers is not in the document]
Comment (brunoais) [reply src/lib/launch.ts:171]: Agh! Screwed up! Got distracted. Didn't pay enough attention. Will fix.

File: src/renpy_warp.rpe.py [hunk 5]
```diff
@@ -17,6 +17,7 @@
 
 enabled = bool(os.getenv("WARP_ENABLED"))
 port = os.getenv("WARP_WS_PORT")
+NONCE = os.getenv("WARP_WS_NONCE")
```
[the remark this answers is not in the document]
Comment (brunoais) [reply src/renpy_warp.rpe.py:20]: The convention is that all caps are constants. But they are constants so, I'm taking that.

File: src/renpy_warp.rpe.py [hunk 6]
```diff
@@ -15,8 +15,13 @@
 
 import renpy  # type: ignore
 
-enabled = bool(os.getenv("WARP_ENABLED"))
-port = os.getenv("WARP_WS_PORT")
+ENABLED = bool(os.getenv("WARP_ENABLED"))
+PORT = os.getenv("WARP_WS_PORT")
+NONCE = os.getenv("WARP_WS_NONCE")
+
+logHere = open("/tmp/socketLog/log.txt", mode="w", buffering=1)
```
Comment (furudean) [inline src/renpy_warp.rpe.py:22]: do we want to keep this? this breaks if `/tmp/socketLog/` doesn't already exist

Comment (furudean) [reply src/renpy_warp.rpe.py:22]: i'm also worried about cross platform compatibility, and not closing the file after you've opened it for writing

Comment (brunoais) [reply src/renpy_warp.rpe.py:22]: That wasn't supposed to have been committed... 🤦I thought I stashed that sorry. Thanks for spotting it.

**Review by furudean — APPROVED** [review]
i've tested that this works. let's merge

**Pull request merged**